iothub_client/inc/iothub_client_core_common.h

```diff
@@ -149,14 +149,56 @@ extern "C"
 
     MU_DEFINE_ENUM_WITHOUT_INVALID(DEVICE_TWIN_UPDATE_STATE, DEVICE_TWIN_UPDATE_STATE_VALUES);
 
+/**
+ * @brief  GET twin request options
+ */
+typedef struct IOTHUB_TWIN_REQUEST_OPTIONS_TAG
+{
+    /** @brief  True if the device's current version is set. If this option is set, `current_version`
+     *          must also be set by the application. If this option is not set, the IoT Hub's
+     *          current version of the requested twin section will be sent back.
+     * @note See @c current_version .
+     **/
+    bool is_current_version_set;
+
+    /** @brief  The device's current version can be used to request a twin section. If the IoT Hub's
+     *          current version matches this value, no payload will be sent. Otherwise, the IoT
+     *          Hub's current version of the requested twin section will be sent back.
+     * @note  Valid range is [1, 9223372036854775807].
+     * @note  This property is optional. See @c is_current_version_set .
+     **/
+    int64_t current_version;
+
+}IOTHUB_TWIN_REQUEST_OPTIONS;
+
+/**
+ * @brief  Get twin response
+ */
+typedef struct IOTHUB_TWIN_RESPONSE_TAG
+{
+    /** @brief True if status has been set from the response. **/
+    bool is_status_set;
+
+    /** @brief If @c is_status_set is true, the status set from the response. **/
```

Comment:
nit here and line 189
```suggestion
    /** @brief The status set from the response (if @c is_status_set is true). **/
```